src/Help/TicketSystem/TicketRepository.php

```diff
@@ -0,0 +1,342 @@
+            ->toArray();
+    }
+
+    private function createObject(array $data): Ticket
+    {
+        $ticket = new Ticket();
+
+        $ticket->id = (int) $data['id'];
+        $ticket->solution = $data['solution'];
+        $ticket->status = $data['status'];
+        $ticket->timestamp = (int) $data['timestamp'];
+        $ticket->adminComment = $data['admin_comment'];
+        $ticket->catId = (int) $data['cat_id'];
+        $ticket->userId = (int) $data['user_id'];
+        $ticket->adminId = (int) $data['admin_id'];
+
+        return $ticket;
+    }
+
+    public function persist(Ticket &$ticket): bool
+    {
+        assert(isset(Ticket::STATUS_ITEMS[$ticket->status]), 'Invalid ticket status');
+        assert(isset(Ticket::SOLUTION_ITEMS[$ticket->solution]), 'Invalid ticket solution');
+
+        $ticket->timestamp = time();
+
+        if ($ticket->id > 0) {
+            $affected = $this->createQueryBuilder()
+                ->update('tickets')
+                ->set('status', ':status')
+                ->set('solution', ':solution')
+                ->set('cat_id', ':cat_id')
+                ->set('admin_id', ':admin_id')
+                ->set('admin_comment', ':admin_comment')
+                ->set('timestamp', ':timestamp')
+                ->where('id = :ticket_id')
+                ->setParameters([
+                    'ticket_id' => $ticket->id,
+                    'status' => $ticket->status,
+                    'solution' => $ticket->solution,
+                    'cat_id' => $ticket->catId,
+                    'admin_id' => $ticket->adminId,
+                    'admin_comment' => $ticket->adminComment,
+                    'timestamp' => $ticket->timestamp,
+                ])
+                ->execute();
+            return (int) $affected > 0;
+        }
+        $this->createQueryBuilder()
+            ->insert('tickets')
+            ->values([
+                'status' => ':status',
+                'solution' => ':solution',
+                'user_id' => ':user_id',
+                'cat_id' => ':cat_id',
+                'admin_id' => ':admin_id',
+                'admin_comment' => ':admin_comment',
+                'timestamp' => ':timestamp',
+            ])
+            ->setParameters([
+                'status' => $ticket->status,
+                'solution' => $ticket->solution,
+                'user_id' => $ticket->userId,
+                'cat_id' => $ticket->catId,
+                'admin_id' => $ticket->adminId ?? 0,
+                'admin_comment' => $ticket->adminComment,
+                'timestamp' => $ticket->timestamp,
+            ])
+            ->execute();
+        $ticket->id = (int) $this->getConnection()->lastInsertId();
+        return true;
+    }
+
+    public function create(int $userId, int $catId, string $message): Ticket
+    {
+        $ticket = new Ticket();
+        $ticket->solution = 'open';
+        $ticket->status = 'new';

```

Comment:
Should move 'open' and 'new' to constants in the ticket class

Reply:
Done 331290b